lambda/input/src/main/scala/InputLambdaMain.scala

```diff
@@ -336,6 +336,13 @@ class InputLambdaMain extends RequestHandler[S3Event, Unit] with DocId with Zone
     })
   }
 
+   def getHeadObjectRequest(rec:S3EventNotification.S3EventNotificationRecord, path: String):HeadObjectRequest = {
+    if (rec.getS3.getObject.getVersionId != null) {
```

Comment:
We need to check for "empty-string" as well.
A more Scala-y way to do it, including the empty-string check, would be:

```scala
def getHeadObjectRequest(rec:S3EventNotification.S3EventNotificationRecord, path: String):HeadObjectRequest = {
  val baseReq = HeadObjectRequest.builder().bucket(rec.getS3.getBucket.getName).key(path)
  getObjectVersion(rec) match {
    case None=>baseReq.build
    case Some(versionId)=>baseReq.versionId(versionId).build
  }
}
```

Reply:
Done.